diff -Naur a/bin/atomenv.csh b/bin/atomenv.csh
--- a/bin/atomenv.csh	2014-01-31 09:35:44.000000000 +0100
+++ b/bin/atomenv.csh	2014-01-31 10:05:11.000000000 +0100
@@ -3,13 +3,13 @@
 set PLATFORM=`uname -s`
 set called=($_)
 
-if ( "$called" != "" ) then 
-   set script_dir=`dirname $called[2]`
-else           
-   set script_dir=`dirname $0`
-endif
+#if ( "$called" != "" ) then 
+#   set script_dir=`dirname $called[2]`
+#else           
+#   set script_dir=`dirname $0`
+#endif
 
-set ATOMCFG=` cd ${script_dir} && pwd `/atom-config
+set ATOMCFG=@prefix@/bin/atom-config
 
 
 setenv ATOM_PATH `${ATOMCFG} --prefix`
diff -Naur a/bin/atomenv.sh b/bin/atomenv.sh
--- a/bin/atomenv.sh	2014-01-31 09:35:44.000000000 +0100
+++ b/bin/atomenv.sh	2014-01-31 10:05:32.000000000 +0100
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 PLATFORM=`uname -s`
-ATOMCFG=$(cd "$( dirname "$0" )" && pwd)/atom-config
+ATOMCFG=@prefix@/bin/atom-config
 export ATOM_PATH=`${ATOMCFG} --prefix`
 
 
