diff -Naur a/CMakeLists.txt c/CMakeLists.txt
--- a/CMakeLists.txt	2014-01-15 13:18:45.000000000 +0100
+++ c/CMakeLists.txt	2014-01-16 20:28:59.000000000 +0100
@@ -151,16 +151,16 @@
 include( UseCython )
 
 #Doxygen support
-find_package(Doxygen)
-if(DOXYGEN_FOUND)
-configure_file(${CMAKE_CURRENT_SOURCE_DIR}/atom.doxy.in ${CMAKE_CURRENT_BINARY_DIR}/atom.doxy ESCAPE_QUOTES @ONLY)
-add_custom_target(doc
-${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/atom.doxy
-WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-COMMENT "Generating API documentation with Doxygen" VERBATIM
-)
-install(DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/html" DESTINATION share OPTIONAL)
-endif(DOXYGEN_FOUND)
+#find_package(Doxygen)
+#if(DOXYGEN_FOUND)
+#configure_file(${CMAKE_CURRENT_SOURCE_DIR}/atom.doxy.in ${CMAKE_CURRENT_BINARY_DIR}/atom.doxy ESCAPE_QUOTES @ONLY)
+#add_custom_target(doc
+#${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/atom.doxy
+#WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+#COMMENT "Generating API documentation with Doxygen" VERBATIM
+#)
+#install(DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/html" DESTINATION share OPTIONAL)
+#endif(DOXYGEN_FOUND)
 
 add_subdirectory(src)
 add_subdirectory(Analyses)
