diff -rupN a/cmake/modules/SetUpMacOS.cmake b/cmake/modules/SetUpMacOS.cmake
--- a/cmake/modules/SetUpMacOS.cmake	2015-01-12 15:26:28.000000000 +0100
+++ b/cmake/modules/SetUpMacOS.cmake	2015-01-30 16:04:05.000000000 +0100
@@ -12,11 +12,11 @@ set(CMAKE_PREFIX_PATH ${CMAKE_PREFIX_PAT
 #---------------------------------------------------------------------------------------------------------
 
 if (CMAKE_SYSTEM_NAME MATCHES Darwin)
-  EXECUTE_PROCESS(COMMAND sw_vers "-productVersion"
+  EXECUTE_PROCESS(COMMAND ${CMAKE_SW_VERS} "-productVersion"
                   COMMAND cut -d . -f 1-2
                   OUTPUT_VARIABLE MACOSX_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
   MESSAGE(STATUS "Found a Mac OS X System ${MACOSX_VERSION}")
-  EXECUTE_PROCESS(COMMAND sw_vers "-productVersion"
+  EXECUTE_PROCESS(COMMAND ${CMAKE_SW_VERS} "-productVersion"
                   COMMAND cut -d . -f 2
                   OUTPUT_VARIABLE MACOSX_MINOR OUTPUT_STRIP_TRAILING_WHITESPACE)
 
