diff -rupN a/cmake/modules/RootNewMacros.cmake b/cmake/modules/RootNewMacros.cmake
--- a/cmake/modules/RootNewMacros.cmake	2015-06-02 17:02:58.000000000 +0900
+++ b/cmake/modules/RootNewMacros.cmake	2015-06-09 18:35:22.000000000 +0900
@@ -248,6 +248,7 @@ function(ROOT_GENERATE_DICTIONARY dictio
   else()
     set(includedirs -I${CMAKE_CURRENT_SOURCE_DIR}/inc)
   endif()
+  set(includedirs ${includedirs} -I${CMAKE_LIBC_DIR}/include)
   foreach( d ${incdirs})
    set(includedirs ${includedirs} -I${d})
   endforeach()
