diff -rupN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2015-06-02 17:02:58.000000000 +0900
+++ b/CMakeLists.txt	2015-06-09 18:38:30.000000000 +0900
@@ -138,7 +138,7 @@ if(roofit)
 endif()
 
 ROOT_ADD_TEST_SUBDIRECTORY(test)
-ROOT_ADD_TEST_SUBDIRECTORY(tutorials)
+# ROOT_ADD_TEST_SUBDIRECTORY(tutorials)
 
 #---Global PCH-----------------------------------------------------------------------------------
 get_property(__allTargets GLOBAL PROPERTY ROOT_DICTIONARY_TARGETS)
@@ -165,22 +165,22 @@ install(FILES ${CMAKE_BINARY_DIR}/etc/al
 install(DIRECTORY ${CMAKE_BINARY_DIR}/etc/dictpch DESTINATION ${CMAKE_INSTALL_SYSCONFDIR})
 
 #---hsimple.root---------(use the executable for clearer dependencies and proper return code)---
-if(NOT gnuinstall)   # TODO hsimple.root for gnuinstall
-  add_custom_target(hsimple ALL DEPENDS tutorials/hsimple.root)
-  add_dependencies(hsimple onepcm)
-  ROOT_EXECUTABLE(hsimple.exe test/hsimple.cxx NOINSTALL LIBRARIES RIO Tree Hist)
-  if(WIN32)
-    add_custom_command(OUTPUT tutorials/hsimple.root
-                       COMMAND set PATH=${CMAKE_RUNTIME_OUTPUT_DIRECTORY} && $<TARGET_FILE:hsimple.exe>
-                       WORKING_DIRECTORY tutorials DEPENDS Cling)
-  else()
-    add_custom_command(OUTPUT tutorials/hsimple.root
-                       COMMAND ${ld_library_path}=${CMAKE_LIBRARY_OUTPUT_DIRECTORY}:$ENV{${ld_library_path}}
-                               $<TARGET_FILE:hsimple.exe>
-                       WORKING_DIRECTORY tutorials DEPENDS Cling)
-  endif()
-  install(FILES ${CMAKE_BINARY_DIR}/tutorials/hsimple.root DESTINATION ${CMAKE_INSTALL_TUTDIR} COMPONENT tests)
-endif()
+# if(NOT gnuinstall)   # TODO hsimple.root for gnuinstall
+#   add_custom_target(hsimple ALL DEPENDS tutorials/hsimple.root)
+#   add_dependencies(hsimple onepcm)
+#   ROOT_EXECUTABLE(hsimple.exe test/hsimple.cxx NOINSTALL LIBRARIES RIO Tree Hist)
+#   if(WIN32)
+#     add_custom_command(OUTPUT tutorials/hsimple.root
+#                        COMMAND set PATH=${CMAKE_RUNTIME_OUTPUT_DIRECTORY} && $<TARGET_FILE:hsimple.exe>
+#                        WORKING_DIRECTORY tutorials DEPENDS Cling)
+#   else()
+#     add_custom_command(OUTPUT tutorials/hsimple.root
+#                        COMMAND ${ld_library_path}=${CMAKE_LIBRARY_OUTPUT_DIRECTORY}:$ENV{${ld_library_path}}
+#                                $<TARGET_FILE:hsimple.exe>
+#                        WORKING_DIRECTORY tutorials DEPENDS Cling)
+#   endif()
+#   install(FILES ${CMAKE_BINARY_DIR}/tutorials/hsimple.root DESTINATION ${CMAKE_INSTALL_TUTDIR} COMPONENT tests)
+# endif()
 
 #---version--------------------------------------------------------------------------------------
 add_custom_target(version COMMAND ${CMAKE_SOURCE_DIR}/build/unix/makeversion.sh ${CMAKE_BINARY_DIR}
