diff -rupN a/interpreter/llvm/src/cmake/config-ix.cmake b/interpreter/llvm/src/cmake/config-ix.cmake
--- a/interpreter/llvm/src/cmake/config-ix.cmake	2015-01-17 09:47:04.000000000 +0100
+++ b/interpreter/llvm/src/cmake/config-ix.cmake	2015-03-25 21:48:20.000000000 +0100
@@ -321,10 +321,12 @@ endif()
 
 # By default, we target the host, but this can be overridden at CMake
 # invocation time.
-include(GetHostTriple)
-get_host_triple(LLVM_INFERRED_HOST_TRIPLE)
+#include(GetHostTriple)
+#get_host_triple(LLVM_INFERRED_HOST_TRIPLE)
 
-set(LLVM_HOST_TRIPLE "${LLVM_INFERRED_HOST_TRIPLE}" CACHE STRING
+#set(LLVM_HOST_TRIPLE "${LLVM_INFERRED_HOST_TRIPLE}" CACHE STRING
+#    "Host on which LLVM binaries will run")
+set(LLVM_HOST_TRIPLE "x86_64-apple-darwin11.4.2" CACHE STRING
     "Host on which LLVM binaries will run")
 
 # Determine the native architecture.
