diff -Naur a/configure b/configure
--- a/configure	2013-11-19 16:49:10.000000000 +0100
+++ b/configure	2014-01-13 22:30:53.000000000 +0100
@@ -227,6 +227,9 @@
 TABLE            \
 XMLDIR           \
 ROOTDICTTYPE     \
+X11              \
+XPM
+XEXT
 "
 configenvvars=
 for c in $envvars ; do
@@ -1347,6 +1350,7 @@
              arch=macosx64
           fi
        fi
+       @patchDarwin64@
     fi
     if [ "$arch" = "solarisCC5" -a "$chip" = "i86pc" ]; then
        if `isainfo -b | grep "64" > /dev/null  2>&1` ; then
@@ -2518,11 +2522,7 @@
 # Check for X11 and Xpm library (for X11 use always shared lib)
 #
 if test ! "x$enable_x11" = "xno" ; then
-    check_library "libX11" "yes" "$x11libdir" \
-        ${finkdir:+$finkdir/lib} \
-        /usr/lib /usr/X11R6/lib /usr/lib/X11 /usr/openwin/lib \
-        /usr/local/lib /usr/local/lib/X11 /usr/local/X11R6/lib \
-        /usr/X11/lib /usr/lib/X11R5
+    check_library "libX11" "yes" "$x11libdir" "$x11dir/lib" "x11dir/lib"
     x11libdir=$found_dir
 
     if test "x$found_lib" = "x" ; then
@@ -2531,11 +2531,7 @@
         exit 1
     fi
 
-    check_header "X11/Xlib.h" "" \
-        ${finkdir:+$finkdir/include} \
-        /usr/include /usr/X11R6/include /usr/include/X11 /usr/openwin/include \
-        /usr/local/include /usr/local/include/X11 /usr/local/X11R6/include \
-        /usr/X11/include /usr/include/X11R5
+    check_header "X11/Xlib.h" "" "$x11dir/include"
     x11incdir=$found_dir
 
     if test "x$found_dir" = "x" ; then
@@ -2544,12 +2540,7 @@
         exit 1
     fi
 
-    check_header "X11/xpm.h" "" \
-        $XPM ${XPM:+$XPM/include} \
-        ${finkdir:+$finkdir/include} \
-        /usr/include /usr/X11R6/include /usr/include/X11 /usr/openwin/include \
-        /usr/local/include /usr/local/include/X11 /usr/local/X11R6/include \
-        /usr/X11/include /usr/include/X11R5
+    check_header "X11/xpm.h" "" "$XPM/include"
 
     if test "x$found_dir" = "x" ; then
         result "`basename $0`: xpm headers MUST be installed"
@@ -2557,11 +2548,7 @@
         exit 1
     fi
 
-    check_header "X11/Xft/Xft.h" "" \
-        ${finkdir:+$finkdir/include} \
-        /usr/include /usr/X11R6/include /usr/include/X11 /usr/openwin/include \
-        /usr/local/include /usr/local/include/X11 /usr/local/X11R6/include \
-        /usr/X11/include /usr/include/X11R5
+    check_header "X11/Xft/Xft.h" "" "$Xftdir/include"
 
     if test "x$found_dir" = "x" ; then
         result "`basename $0`: Xft headers MUST be installed"
@@ -2570,11 +2557,7 @@
     fi
 
     if test ! "x$arch" = "xaix5" ; then
-        check_header "X11/extensions/shape.h" "" \
-            ${finkdir:+$finkdir/include} \
-            /usr/include /usr/X11R6/include /usr/include/X11 \
-            /usr/openwin/include /usr/local/include /usr/local/include/X11 \
-            /usr/local/X11R6/include /usr/X11/include /usr/include/X11R5
+        check_header "X11/extensions/shape.h" "" "$XEXT/include"  
 
         if test "x$found_dir" = "x" ; then
             result "`basename $0`: X11/extensions/shape.h header MUST be installed"
@@ -2583,12 +2566,7 @@
         fi
     fi
 
-    check_library "libXpm" "$enable_shared" "$xpmlibdir" \
-        $XPM ${XPM:+$XPM/lib} \
-        ${finkdir:+$finkdir/lib} \
-        /usr/local/lib /usr/lib /usr/lib/X11 \
-        /usr/local/lib/X11 /usr/X11R6/lib /usr/local/X11R6/lib \
-        /usr/X11/lib
+    check_library "libXpm" "$enable_shared" "$xpmlibdir" $XPM ${XPM:+$XPM/lib} 
     xpmlib=$found_lib
     xpmlibdir=$found_dir
 
@@ -2598,11 +2576,7 @@
         exit 1
     fi
 
-    check_library "libXft" "$enable_shared" "$xftlibdir" \
-        ${finkdir:+$finkdir/lib} \
-        /usr/local/lib /usr/lib /usr/lib/X11 \
-        /usr/local/lib/X11 /usr/X11R6/lib /usr/local/X11R6/lib \
-        /usr/X11/lib
+    check_library "libXft" "$enable_shared" "$xftlibdir" "$Xftdir/lib"
 
     if test "x$found_lib" = "x" ; then
         result "`basename $0`: libXft MUST be installed"
@@ -2616,11 +2590,7 @@
         exit 1
     fi
 
-    check_library "libXext" "$enable_shared" "$xextlibdir" \
-        ${finkdir:+$finkdir/lib} \
-        /usr/local/lib /usr/lib /usr/lib/X11 \
-        /usr/local/lib/X11 /usr/X11R6/lib /usr/local/X11R6/lib \
-        /usr/X11/lib
+    check_library "libXext" "$enable_shared" "$XEXT/lib" "$xextlibdir" "$Xextdir/lib"
 
     if test "x$found_lib" = "x" ; then
         result "`basename $0`: libXext MUST be installed"
@@ -4334,22 +4304,15 @@
     # for a system library, then see if we have various headers needed.
     if test "x$enable_builtin_afterimage" = "xyes" && test ! "x$enable_cocoa" = "xyes"; then
        check_header "jpeglib.h" "" \
-          $ASIMAGE ${ASIMAGE:+$ASIMAGE/include} \
-          ${finkdir:+$finkdir/include} \
-          /usr/local/include /usr/include /opt/include
+          $ASIMAGE ${ASIMAGE:+$ASIMAGE/include} 
        asjpegincdir=$found_dir
        check_header "png.h" "" \
           $ASIMAGE ${ASIMAGE:+$ASIMAGE/include} \
-          ${ASPNG:+$ASPNG/include} \
-          ${finkdir:+$finkdir/include} \
-          /usr/local/include /usr/X11/include /usr/include \
-          /usr/local/include/libpng /opt/include
+          ${ASPNG:+$ASPNG/include} $libpng/include
        aspngincdir=$found_dir
        if test ! "x$enable_astiff" = "xno" ; then
            check_header "tiffio.h" "" \
                $ASIMAGE ${ASIMAGE:+$ASIMAGE/include} \
-               ${finkdir:+$finkdir/include} \
-               /usr/local/include /usr/include /opt/include
            astiffincdir=$found_dir
        else
            astiffincdir="--with-tiff=no"
@@ -4368,9 +4331,7 @@
        fi
        for k in $aslibs ; do
            check_library $k "$enable_shared" "" \
-               $ASIMAGE ${ASIMAGE:+$ASIMAGE/lib} ${ASPNG:+$ASPNG/lib} \
-               ${finkdir:+$finkdir/lib} \
-               /usr/local/lib /usr/X11/lib /usr/lib /opt/lib
+               $ASIMAGE ${ASIMAGE:+$ASIMAGE/lib} ${ASPNG:+$ASPNG/lib} $libpng/lib 
            if test "x$k" = "xlibz" ; then
                found_libz=$found_lib
                found_dirz=$found_dir
@@ -6638,7 +6599,7 @@
     if test "x$bindir" = "x";       then bindir=$prefix/bin; fi
     if test "x$libdir" = "x";       then libdir=$prefix/lib/root; fi
     if test "x$incdir" = "x";       then incdir=$prefix/include/root; fi
-    if test "x$etcdir" = "x";       then etcdir=/etc/root; fi
+    if test "x$etcdir" = "x";       then etcdir=$prefix/etc; fi
     if test "x$mandir" = "x";       then mandir=$prefix/share/man/man1; fi
     if test "x$datadir" = "x";      then datadir=$prefix/share/root; fi
     if test "x$macrodir" = "x";     then macrodir=$datadir/macros; fi
