diff -rupN a/Makefile b/Makefile
--- a/Makefile	2015-06-12 10:47:16.000000000 +0900
+++ b/Makefile	2015-06-16 14:24:02.000000000 +0900
@@ -18,8 +18,8 @@
 #
 # GNU C Compiler:
 # ===============
- CC		= gcc
- TSIL_OPT	= -O3 -funroll-loops 
+# CC		= gcc
+# TSIL_OPT	= -O3 -funroll-loops 
 #
 # Intel C compiler:
 # =================
@@ -28,8 +28,8 @@
 #
 # General C compiler:
 # ===================
-# CC	 	= cc
-# TSIL_OPT 	= -O3
+CC	 	= cc
+TSIL_OPT 	= -O3
 #
 ################### INSTALLATION DIRECTORIES #######################
 #
